wordpress-mobile/WordPress-Android
# Issue/13326 fetch fix threats status

Parent #13326

This PR fetches fix threats status after fix threats action is started and handles result from `FetchFixThreatsStatusUseCase`.
Child PR for `FetchFixThreatsStatusUseCase`: #13797

**Threat Fixing Flow**

https://user-images.githubusercontent.com/1405144/104889072-a3e9b880-5993-11eb-93ed-3306ff2b57e0.mov

https://user-images.githubusercontent.com/1405144/104892751-8ff48580-5998-11eb-8cfc-b7fec137fd58.mp4

**To test**
Prerequisite:

1. Make sure both Scan feature flag and MySiteImprovements flag are set to on in the App settings.
2. You have access to WP.com account with a site having scan capability and threats (e.g. https://pressable-jetpack-daily-scan.mystagingwebsite.com). Server credentials should be set on the site.
3. I've added a threat to the site. In case you run out of threats, re-install bad calendar plugin from [here](https://jetpackp2.files.wordpress.com/2020/05/calendar-1.zip) and re-run scan.

- Login to the app using above WP.com account and select the site on the My Site tab
- Click scan menu item on My Site
- Click "Fix x threats" button on scan state screen with threats
- Click ok on the confirmation dialog
- Notice that
   - Indeterminate progress bar is shown and action buttons are disabled while threat is fixing
   - Action buttons are re-enabled when threat fixing completes
   - Success/ error message is shown when threat fixing completes

(Tests covered in `ScanViewModelTest`)

Notes:

1. Ignore all styling.
2. Server credentials check is not handled.
3. While the scan design i5 (`pcdRpT-8X`) displays an incremental progress bar, an indeterminate progress bar is shown as we do not get threat fixing progress updates from the API.
4. Fix threats status progress bar is shown only while user doesn't leave scan state screen. Once user leaves and re-enters scan state screen from the My Site screen, latest scan state is fetched and shown. It will display scanning state incremental progress bar if this latest state is found to be scanning. Note that web version doesn't display "fix threats status progress bar". Suggestions are welcome to streamline this flow.

Merge Instructions

1. Make sure child PR #13797 is merged to develop
2. Remove the "Not Ready for Merge label"
3. Merge this PR

PR submission checklist:
